### PR TITLE
[Prototype] Support for DEFAULT column parsing and analysis

### PR DESCRIPTION
**Pull request by dtenedor**

### CODE REVIEW

File: .idea/vcs.xml [hunk 1]
```diff
@@ -1,36 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<!--
```
Comment (gengliangwang) [inline .idea/vcs.xml:2]: nit: unnecessary change

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/DefaultColumns.scala [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+import org.apache.spark.sql.catalyst.optimizer.ConstantFolding
+import org.apache.spark.sql.catalyst.parser.{CatalystSqlParser, ParseException}
+import org.apache.spark.sql.catalyst.plans.logical._
+import org.apache.spark.sql.connector.catalog._
+import org.apache.spark.sql.types._
+
+/**
+ * This class contains logic for processing DEFAULT columns in statements such as CREATE TABLE.
+ */
+case class DefaultColumns(catalogManager: CatalogManager) {
```
Comment (gengliangwang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/DefaultColumns.scala]: It seems that `catalogManager` is not used.  Shall we make it 
```
object DefaultColumnUtils
``` 
?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala [hunk 3]
```diff
@@ -368,12 +369,13 @@ case class PreprocessTableCreation(sparkSession: SparkSession) extends Rule[Logi
  * table. It also does data type casting and field renaming, to make sure that the columns to be
  * inserted have the correct data type and fields have the correct names.
  */
-object PreprocessTableInsertion extends Rule[LogicalPlan] {
+case class PreprocessTableInsertion(catalogManager: CatalogManager) extends Rule[LogicalPlan] {
```
Comment (gengliangwang) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala]: catalogManager is not used

File: .idea/vcs.xml (hunk 1 above)
Comment (dtenedor) [reply .idea/vcs.xml:2]: Yeah I accidentally uploaded the changed vcs.xml file :) I'll skip this in the real PRs

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/DefaultColumns.scala (hunk 2 above)
Comment (dtenedor) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/DefaultColumns.scala]: SG, done.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala (hunk 3 above)
Comment (dtenedor) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/rules.scala]: Done.

**Comment by dtenedor** [comment]
I'm going to drop this now since the merged PRs have caught up with the prototype :)
